scripts/logfire_query.py

```
@@ -0,0 +1,142 @@
+    return run_query(sql)
+
+# Query logs for a specific project
+def query_project(project_name="vibemachine", limit=5):
+    print(f"=== Last {limit} logs from '{project_name}' ===\n")
+    sql = f"""
+        SELECT start_timestamp, level, span_name, message, attributes
+        FROM records
+        WHERE service_name = '{project_name}'
+        ORDER BY start_timestamp DESC
+        LIMIT {limit}
+    """
```

Ruff (0.14.5)

Possible SQL injection vector through string-based query construction

(S608)